neutrino.go

```diff
@@ -815,6 +834,46 @@ func NewChainService(cfg Config) (*ChainService, error) {
 		return nil, err
 	}
 	s.blockManager = bm
+
+	if cfg.BlkHdrSideload != nil {
+		blockHdrChkptMgr := NewblockHeaderCheckpoints(s.chainParams)
+		blockHeaderWriter := &blkHeaderWriter{
+			store:     s.BlockHeaders,
+			tipHeight: s.blockManager.headerTip,
+		}
+
+		loader := sideload.LoaderInternal[*wire.BlockHeader]{
+			HeaderWriter: blockHeaderWriter,
+			HeaderValidator: &blockHeaderValidator{
+				BlockHeaderCheckpoints: blockHdrChkptMgr,
+				headerList:             bm.headerList,
+				minRetargetTimespan:    bm.minRetargetTimespan,
+				maxRetargetTimespan:    bm.maxRetargetTimespan,
+				blocksPerRetarget:      bm.blocksPerRetarget,
+				ChainParams:            s.chainParams,
+				TimeSource:             s.timeSource,
+				nextCheckpoint:         bm.nextCheckpoint,
+				store:                  s.BlockHeaders,
+			},
+			SkipVerify:    cfg.BlkHdrSideload.SkipVerify,
+			Chkpt:         blockHdrChkptMgr,
+			SideloadRange: SideloadRange,
+		}
```

Comment:
Please note that I have not modified the blockmanager yet, if my approach is okay, the PR that I will submit would not have these.